src/fu-device.h

```diff
@@ -170,6 +177,9 @@ FwupdRelease	*fu_device_get_release_default		(FuDevice	*device);
 gboolean	 fu_device_write_firmware		(FuDevice	*device,
 							 GBytes		*fw,
 							 GError		**error);
+gboolean	 fu_device_check_firmware		(FuDevice	*device,
```

Comment:
I think the functionality is more like  `fu_device_check_firmware_blob`. or `fu_device_validate_firmware_blob`  `fu_device_check_firmware` has an implicit connotation of checking firmware already on the `fu_device`.

Reply:
I guess `fu_device_validate_firmware()` or `fu_device_check_firmware_blob()` make sense -- of course the other thing we could do is have a `GBytes *fu_device_prepare_firmware(FuDevice, GBytes, GError)` function which could have an optional klass device-specific hook for doing the IHEX->BIN at the same time. Then that way we could just put it in write_firmware() in one central place and the two plugins that do odd unpacking just need a bit of code refactored.